cli-plugins/manager/cobra.go

```diff
@@ -136,12 +137,41 @@ func appendPluginResourceAttributesEnvvar(env []string, cmd *cobra.Command, plug
 	if attrs := getPluginResourceAttributes(cmd, plugin); attrs.Len() > 0 {
 		// values in environment variables need to be in baggage format
 		// otel/baggage package can be used after update to v1.22, currently it encodes incorrectly
-		attrsSlice := make([]string, attrs.Len())
+		// Construct baggage members for each of the attributes.
+		// Ignore any failures as these aren't significant and
+		// represent an internal issue.
+		var b baggage.Baggage
```

Comment:
wouldn't `b` be nil on the first iteration?

Why not have a slice of members and then add it to the  baggage afterwards.

```go
var members []baggage.Member
for iter := attrs.Iter(); iter.Next(); {
  m, err := baggage.NewMemberRaw(string(attr.Key), attr.Value.AsString())
  if err != nil {
    continue
  }
  members = append(members, m)
}
b := baggage.New(members...)

Reply:
Thank you! it was just quite difficult to read, since we were assigning to a variable in the outer-scope on every iteration - just looked strange. 

```go
for //
newB, err := b.SetMember(m)
// handle err
b = newB
```